frontend/src/auth/AuthProvider.js

```diff
@@ -2,18 +2,23 @@ import AuthContext from './AuthContext'
 import { useContext, useState } from 'react'
 import jwt from 'jsonwebtoken'
 import { getToken } from '../services/api-service'
+import { useHistory } from 'react-router-dom'
 
 export default function AuthProvider({ children }) {
   const [token, setToken] = useState()
   const claims = jwt.decode(token)
+  const history = useHistory()
 
   const user = claims && {
     username: claims.sub,
     role: claims.role,
   }
   const login = credentials => getToken(credentials).then(setToken)
 
-  const logout = () => setToken()
+  const logout = () => {
+    setToken()
+    history.push('/login')
+  }
```

Comment:
Du kannst auch in der app. js eine Route mit to="/" component={login} schreiben um immer dorthin geleitete zu werden.

Reply:
ich habe in der App.js eine andere Route mit "/", da ich noch einen anderen frei zugänglichen Bereich haben möchte. Nur nach dem logout möchte ich wieder zum login